ui/tutorial-renderer/src/main/java/org/cru/godtools/tutorial/animation/TextViewAnimation.kt

```diff
@@ -0,0 +1,40 @@
+package org.cru.godtools.tutorial.animation
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import androidx.annotation.StringRes
+import androidx.appcompat.widget.AppCompatTextView
+
+private const val DELAY_DURATION = 3000L
+private const val FADE_DURATION = 1000L
+
+fun AppCompatTextView.animateToNextText(@StringRes animateNextText: Int) {
+    alpha = 1f
+    val fadeOutAnimator = ObjectAnimator.ofFloat(this, "alpha", 1f, 0f).apply {
+        duration = FADE_DURATION
+        setTextAfterAnimation(animateNextText, this@animateToNextText)
+    }
+    val fadeInAnimator = ObjectAnimator.ofFloat(this, "alpha", 0f, 1f).apply {
+        duration = FADE_DURATION
+    }
+    AnimatorSet().apply {
+        startDelay = DELAY_DURATION
+        play(fadeOutAnimator).before(fadeInAnimator)
+        start()
+    }
+}
+
+private fun ObjectAnimator.setTextAfterAnimation(@StringRes animateNextText: Int, textView: AppCompatTextView) {
+    addListener(object : Animator.AnimatorListener {
```

Comment:
since you are only implementing a single method you should extend `AnimatorListenerAdapter` instead to simplify this code